linux/src/main/java/com/github/weisj/darkmode/platform/linux/gnome/GnomeThemeMonitorService.java

```diff
@@ -0,0 +1,82 @@
+package com.github.weisj.darkmode.platform.linux.gnome;
+
+import com.github.weisj.darkmode.platform.ThemeMonitorService;
+
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import static com.github.weisj.darkmode.platform.linux.gnome.GtkVariants.guessFrom;
+import static java.util.Objects.isNull;
+
+public class GnomeThemeMonitorService implements ThemeMonitorService {
+    private final Logger logger = Logger.getLogger(GnomeThemeMonitorService.class.getName());
+    private boolean initializedGnomeGtkThemeMonitorSuccessfully;
+    private GnomeGtkThemeMonitor gnomeGtkThemeMonitor;
+    private Thread gnomeGtkThemeMonitorThread;
+
+    public GnomeThemeMonitorService() {
+        try {
+            gnomeGtkThemeMonitor = new GnomeGtkThemeMonitor();
+            initializedGnomeGtkThemeMonitorSuccessfully = true;
+        } catch (IOException | InterruptedException e) {
+            initializedGnomeGtkThemeMonitorSuccessfully = false;
+            logger.log(Level.SEVERE, e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public boolean isDarkThemeEnabled() {
+        //TODO: Stop guessing and check the settings when available (like what is mentioned in the GtkVariants class)
+        return gnomeGtkThemeMonitor.readGtkTheme().equals(guessFrom(gnomeGtkThemeMonitor.readGtkTheme()).get("night"));
```

Comment:
Reading the theme twice should be avoided as it starts a new process each time.

Reply:
[Changed and fixed](https://github.com/weisJ/auto-dark-mode/commit/610a6ebc3ab8d8ecb73ba32593a245a2b4502cc8#diff-c8788dcf047c4ef3cb989a2551bb3d9bR10).